### PR TITLE
feat: distribute the package as ESM

### PR DESCRIPTION
**Pull request by mattcosta7**
Adds an esm build to this module

Inherently this will _not_ work in a browser directly without node polyfills, even though we emit proper esm (it does work in environments that add in node polyfills, consumers of msw now should not be impacted by that). This is because it relies on the node `events` library which tsup specifically treats as external, to avoid version issues. It might be interesting to try alternative bundlers which might bundle this into the library or potentially try a more native solution like `EventTarget` which is available in both node and browser runtimes. 

Maybe https://github.com/browserify/events/pull/87 landing would help here, eventually

`@web/dev-server` serves this module properly with this configuration, in a browser

```
import { rollupAdapter } from '@web/dev-server-rollup';
import nodePolyfills from 'rollup-plugin-node-polyfills';

export default {
    port: 1234,
    watch: true,
    nodeResolve: true,
    plugins:[
        rollupAdapter(nodePolyfills()),
    ],
};
```

### CODE REVIEW

File: tsup.config.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import { defineConfig } from 'tsup'
+
+export default defineConfig([
+  {
+    name: 'node',
```
Comment (mattcosta7) [inline tsup.config.ts]:
```suggestion
    name: 'main',
```

**Comment by kettanaito** [comment]
I've resolved the conflicts with the main branch. This change should be ready to go now. Will give it the final review and see the CI results.

**Comment by kettanaito** [comment]
We also don't depend on the `events` anymore, which alleviates some of your concerns about it not being distributed correctly.

**Review by kettanaito — APPROVED** [review]

**Comment by kettanaito** [comment]
## Released: v0.5.0 🎉

This has been released in v0.5.0!

- 📄 [**Release notes**](https://github.com/open-draft/strict-event-emitter/releases/tag/v0.5.0)
- 📦 [npm package](https://www.npmjs.com/package/strict-event-emitter/v/0.5.0)

Make sure to always update to the latest version (`npm i strict-event-emitter@latest`) to get the newest features and bug fixes.

---

_Predictable release automation by [@ossjs/release](https://github.com/ossjs/release)_.

**Pull request merged**